backend/ecs_tasks/delete_files/json_handler.py

```diff
@@ -0,0 +1,66 @@
+import gzip
+import io
+import json
+import logging
+from collections import Counter
+
+import pyarrow as pa
+
+logger = logging.getLogger(__name__)
+
+
+def initialize(input_file, out_stream, compressed):
+    if compressed:
+        bytestream = io.BytesIO(input_file.read())
+        input_file = gzip.GzipFile(None, "rb", fileobj=bytestream)
+    gzip_stream = pa.CompressedOutputStream(out_stream, "gzip") if compressed else None
+    writer = gzip_stream if compressed else out_stream
+    return input_file, writer
+
+
+def find_key(key, obj):
+    """
+    Athena openx SerDe is case insensitive, and converts by default each object's key
+    to a lowercase value: https://docs.aws.amazon.com/athena/latest/ug/json-serde.html
+
+    Here we convert the DataMapper value for the column identifier
+    (for instance, customerid) to the JSON's object key (for instance, customerId).
+    """
+    if not obj:
+        return None
+    for found_key in obj.keys():
+        if key.lower() == found_key.lower():
+            return found_key
+
+
+def delete_matches_from_json_file(input_file, to_delete, compressed=False):
+    deleted_rows = 0
+    with pa.BufferOutputStream() as out_stream:
+        input_file, writer = initialize(input_file, out_stream, compressed)
+        content = input_file.read().decode("utf-8")
```

Comment:
do we need to state UTF8 encoding as a limitation somewhere in the docs or is that overkill?

Reply:
Not sure, sounds like overkill to me? I'm unopinionated - happy to add a note but I didn't see anything similar in Athena and similar tools